### PR TITLE
feat(FluidSearch): adds experimental `FluidSearch`

### LINKED ISSUE #12456: Fluid inputs: Add FluidSearch experimental implementation

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/12456

Adds in the `FluidSearch` component as `unstable`

#### Changelog

**New**
- `unstable__FluidSearch` component
- Styles for `FluidSearch` (`_fluid-search.scss`)
- Stories for `FluidSearch` (`FluidSearch.stories.js`)
- Tests for `FluidSearch`

**Changed**

- Updates index, snapshots

#### Testing / Reviewing

Go to `FluidSearch` and ensure all stories look correct.

### CODE REVIEW

**Review by aagonzales — CHANGES_REQUESTED** [review]
In the disabled state, the label needs to also change to `$text-disabled` and the border-bottom should still be shown but as `$border-subtle`

image


I also don't think search needs the tooltip, I don't think its a feature normally associated with a search field.

image

**Comment by tw15egan** [comment]
@aagonzales updated 👍

File: packages/react/src/components/FluidSearch/FluidSearch.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    </FormContext.Provider>
+  );
+});
+
+FluidSearch.propTypes = {
+  /**
+   * Specify an optional value for the `autocomplete` property on the underlying
+   * `<input>`, defaults to "off"
+   */
+  autoComplete: PropTypes.string,
```
Comment (tay1orjones) [inline packages/react/src/components/FluidSearch/FluidSearch.js:41]: Should we still provide the default within this component? I think without them we won't see the defaults in the storybook documentation and prop tables, etc.

File: packages/react/src/components/FluidSearch/FluidSearch.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+   * Provide an optional placeholder text for the Search.
+   * Note: if the label and placeholder differ,
+   * VoiceOver on Mac will read both
+   */
+  placeholder: PropTypes.string,
+
+  /**
+   * Specify the role for the underlying `<input>`, defaults to `searchbox`
+   */
+  role: PropTypes.string,
```
Comment (tay1orjones) [inline packages/react/src/components/FluidSearch/FluidSearch.js:98]: Same thing here with the defaults.

**Review by tay1orjones — APPROVED** [review]
I made a couple minor comments below. Overall looks great! 🎉

**Review by aagonzales — APPROVED** [review]
Nice, looks good to me now!

**Review by alisonjoseph — APPROVED** [review]

**Comment by tw15egan** [comment]
@tay1orjones added, thanks for the suggestion 👍🏻

**Pull request merged**